Netatmo: partnerdevices connection timeout => thing remaining in INITIALIZING state.

```
---- ERROR https://api.netatmo.net/api/partnerdevices
java.io.IOException: org.apache.oltu.oauth2.common.exception.OAuthSystemException: java.net.ConnectException: Connection timed out: connect
	at io.swagger.client.auth.OAuth.updateAccessToken(OAuth.java:140)
	at io.swagger.client.auth.OAuth.intercept(OAuth.java:86)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:225)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:199)
	at com.squareup.okhttp.Call.execute(Call.java:79)
	at retrofit.client.OkClient.execute(OkClient.java:53)
	at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:326)
	at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240)
	at com.sun.proxy.$Proxy88.partnerdevices(Unknown Source)
	at org.openhab.binding.netatmo.handler.NetatmoBridgeHandler.lambda$0(NetatmoBridgeHandler.java:121)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.oltu.oauth2.common.exception.OAuthSystemException: java.net.ConnectException: Connection timed out: connect
	at io.swagger.client.auth.OAuthOkHttpClient.execute(OAuthOkHttpClient.java:61)
	at org.apache.oltu.oauth2.client.OAuthClient.accessToken(OAuthClient.java:65)
	at org.apache.oltu.oauth2.client.OAuthClient.accessToken(OAuthClient.java:55)
	at org.apache.oltu.oauth2.client.OAuthClient.accessToken(OAuthClient.java:71)
	at io.swagger.client.auth.OAuth.updateAccessToken(OAuth.java:129)
	... 16 more
Caused by: java.net.ConnectException: Connection timed out: connect
	at java.net.DualStackPlainSocketImpl.connect0(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:79)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at com.squareup.okhttp.internal.Platform.connectSocket(Platform.java:107)
	at com.squareup.okhttp.Connection.connect(Connection.java:156)
	at com.squareup.okhttp.Connection.connectAndSetOwner(Connection.java:175)
	at com.squareup.okhttp.OkHttpClient$1.connectAndSetOwner(OkHttpClient.java:120)
	at com.squareup.okhttp.internal.http.HttpEngine.nextConnection(HttpEngine.java:330)
	at com.squareup.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:319)
	at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:241)
	at com.squareup.okhttp.Call.getResponse(Call.java:271)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:228)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:199)
	at com.squareup.okhttp.Call.execute(Call.java:79)
	at io.swagger.client.auth.OAuthOkHttpClient.execute(OAuthOkHttpClient.java:54)
	... 20 more

---- END ERROR
2018-02-18 14:17:43.732 [WARN ] [rappedScheduledExecutorService:67   ] - Scheduled runnable ended with an exceptionjava.lang.NullPointerException: null
	at org.openhab.binding.netatmo.handler.NetatmoBridgeHandler.lambda$0(NetatmoBridgeHandler.java:126)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
```

It leads to a NPE in `scheduleTokenInitAndRefresh` in the catch code and so this code is not rescheduled. Last consequence is that the brigde thing remains in a INITIALIZING state.

As a side note, I don't understand why the API exits now in timeout, it was working well 2 hjours ago !

By the way, the issue is that we should catch this timeout exception and put the thing in a OFFLINE state.

--- a/addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoBridgeHandler.java
+++ b/addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoBridgeHandler.java
@@ -52,6 +52,7 @@ import io.swagger.client.model.NAWebhookCameraEventPerson;
 import io.swagger.client.model.NAWelcomeHomeData;
 import retrofit.RestAdapter.LogLevel;
 import retrofit.RetrofitError;
+import retrofit.RetrofitError.Kind;
 
 /**
  * {@link NetatmoBridgeHandler} is the handler for a Netatmo API and connects it
@@ -119,23 +120,32 @@ public class NetatmoBridgeHandler extends BaseBridgeHandler {
                 getPartnerApi().partnerdevices();
                 connectionSucceed();
             } catch (RetrofitError e) {
-                switch (e.getResponse().getStatus()) {
-                    case 404: // If no partner station has been associated - likely to happen - we'll have this error
-                              // but it means connection to API is OK
-                        connectionSucceed();
-                        break;
-                    case 403: // Forbidden Access maybe too many requests ? Let's wait next cycle
-                        logger.warn("Error 403 while connecting to Netatmo API, will retry in {} s",
-                                configuration.reconnectInterval);
-                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_REGISTERING_ERROR,
-                                "Netatmo Access Forbidden, will retry in " + configuration.reconnectInterval.toString()
-                                        + " seconds.");
-                        break;
-                    default:
-                        logger.error("Unable to connect Netatmo API : {}", e.getMessage(), e);
-                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                                "Unable to connect Netatmo API : " + e.getLocalizedMessage());
-                        return;
+                if (e.getKind() == Kind.NETWORK) {
+                    logger.warn("Network error while connecting to Netatmo API, will retry in {} s",
+                            configuration.reconnectInterval);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                            "Netatmo Access Failed, will retry in " + configuration.reconnectInterval.toString()
+                                    + " seconds.");
+                } else {
+                    switch (e.getResponse().getStatus()) {
+                        case 404: // If no partner station has been associated - likely to happen - we'll have this
+                                  // error
+                                  // but it means connection to API is OK
+                            connectionSucceed();
+                            break;
+                        case 403: // Forbidden Access maybe too many requests ? Let's wait next cycle
+                            logger.warn("Error 403 while connecting to Netatmo API, will retry in {} s",
+                                    configuration.reconnectInterval);
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_REGISTERING_ERROR,
+                                    "Netatmo Access Forbidden, will retry in "
+                                            + configuration.reconnectInterval.toString() + " seconds.");
+                            break;
+                        default:
+                            logger.error("Unable to connect Netatmo API : {}", e.getMessage(), e);
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                    "Unable to connect Netatmo API : " + e.getLocalizedMessage());
+                            return;
+                    }
                 }
             }
             // We'll do this every x seconds to guaranty token refresh

--- a/addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoBridgeHandler.java
+++ b/addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoBridgeHandler.java
@@ -123,9 +123,8 @@ public class NetatmoBridgeHandler extends BaseBridgeHandler {
                 if (e.getKind() == Kind.NETWORK) {
                     logger.warn("Network error while connecting to Netatmo API, will retry in {} s",
                             configuration.reconnectInterval);
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
-                            "Netatmo Access Failed, will retry in " + configuration.reconnectInterval.toString()
-                                    + " seconds.");
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Netatmo Access Failed, will retry in " + configuration.reconnectInterval + " seconds.");
                 } else {
                     switch (e.getResponse().getStatus()) {
                         case 404: // If no partner station has been associated - likely to happen - we'll have this
@@ -136,9 +135,9 @@ public class NetatmoBridgeHandler extends BaseBridgeHandler {
                         case 403: // Forbidden Access maybe too many requests ? Let's wait next cycle
                             logger.warn("Error 403 while connecting to Netatmo API, will retry in {} s",
                                     configuration.reconnectInterval);
-                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_REGISTERING_ERROR,
-                                    "Netatmo Access Forbidden, will retry in "
-                                            + configuration.reconnectInterval.toString() + " seconds.");
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "Netatmo Access Forbidden, will retry in " + configuration.reconnectInterval
+                                            + " seconds.");
                             break;
                         default:
                             logger.error("Unable to connect Netatmo API : {}", e.getMessage(), e);

